### PR TITLE
Make current time

### PR DESCRIPTION
**Pull request by glebdronchenko**

### CODE REVIEW

File: src/components/CurrentTime/CurrentTime.jsx [hunk 1]
```diff
@@ -1,7 +1,22 @@
 import classes from './CurrentTime.module.scss';
+import { useState, useEffect } from 'react';
 
 function CurrentTime() {
-  return <div className={classes.currentTime}>17:00</div>;
+  const [currentTime, setCurrentTime] = useState(new Date().toTimeString());
```
Comment (anton-vlasik) [inline src/components/CurrentTime/CurrentTime.jsx]: I guess you could benefit a little bit here from lazy initializing

File: src/components/CurrentTime/CurrentTime.jsx [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+
+  useEffect(() => {
+    setTimeout(() => {
+      const dateObj = new Date();
```
Comment (anton-vlasik) [inline src/components/CurrentTime/CurrentTime.jsx]: all these foramttings could be a part of a separate utility function

Comment (glebdronchenko) [reply src/components/CurrentTime/CurrentTime.jsx]: Done.

File: src/components/CurrentTime/CurrentTime.jsx (hunk 1 above)
Comment (glebdronchenko) [reply src/components/CurrentTime/CurrentTime.jsx]: Done.

File: src/components/CurrentTime/CurrentTime.jsx [hunk 3]
```diff
@@ -1,7 +1,19 @@
 import classes from './CurrentTime.module.scss';
+import { useState, useEffect } from 'react';
+import { getFormatedCurrentTime } from '../../utils/dateTimeUtils';
 
 function CurrentTime() {
-  return <div className={classes.currentTime}>17:00</div>;
+  const [currentTime, setCurrentTime] = useState(() => getFormatedCurrentTime());
```
Comment (DmitriyKrivovyaz) [inline src/components/CurrentTime/CurrentTime.jsx]: if i am right u can wtite just useState(getFormatedCurrentTima)

**Review by anton-vlasik — APPROVED** [review]

File: src/components/CurrentTime/CurrentTime.jsx (hunk 3 above)
Comment (glebdronchenko) [reply src/components/CurrentTime/CurrentTime.jsx]: No. Every rerender this functions will call twice. You can check it.

Comment (DmitriyKrivovyaz) [reply src/components/CurrentTime/CurrentTime.jsx]: okey, sorry

Comment (AV29) [reply src/components/CurrentTime/CurrentTime.jsx]: Actually I agree with @DmitriyKrivovyaz. You should be able to do that.

File: src/components/CurrentTime/CurrentTime.jsx [hunk 4]
```diff
@@ -1,7 +1,19 @@
 import classes from './CurrentTime.module.scss';
+import { useState, useEffect } from 'react';
+import { getFormatedCurrentTime } from '../../utils/dateTimeUtils';
```
Comment (AV29) [inline src/components/CurrentTime/CurrentTime.jsx]: formatted is spelled with 2 t's

File: src/utils/dateTimeUtils.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+function getFormatedCurrentTime() {
+  const dateObj = new Date();
+  const hours = dateObj.getHours().toString().padStart(2, 0);
+  const minutes = dateObj.getMinutes().toString().padStart(2, 0);
+  const seconds = dateObj.getSeconds().toString().padStart(2, 0);
```
Comment (AV29) [inline src/utils/dateTimeUtils.js]: second parameter for padStart should be of type string

File: src/components/CurrentTime/CurrentTime.jsx (hunk 3 above)
Comment (glebdronchenko) [reply src/components/CurrentTime/CurrentTime.jsx]: Sorry I was wrong) @DmitriyKrivovyaz was right!

File: src/utils/dateTimeUtils.js (hunk 5 above)
Comment (glebdronchenko) [reply src/utils/dateTimeUtils.js]: Done.

File: src/components/CurrentTime/CurrentTime.jsx (hunk 4 above)
Comment (glebdronchenko) [reply src/components/CurrentTime/CurrentTime.jsx]: Done.

**Review by anton-vlasik — APPROVED** [review]

File: src/components/CurrentTime/CurrentTime.jsx [hunk 6]
```diff
@@ -1,7 +1,19 @@
...
 function CurrentTime() {
-  return <div className={classes.currentTime}>17:00</div>;
+  const [currentTime, setCurrentTime] = useState(getFormattedCurrentTime);
+
+  useEffect(() => {
+    setTimeout(() => {
+      const timeString = getFormattedCurrentTime();
+
+      setCurrentTime(timeString);
+    }, 1000);
```
Comment (ximet) [inline src/components/CurrentTime/CurrentTime.jsx]: 1000 it's magic number for me

File: src/utils/dateTimeUtils.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+function getFormattedCurrentTime() {
+  const dateObj = new Date();
+  const hours = dateObj.getHours().toString().padStart(2, '0');
```
Comment (ximet) [inline src/utils/dateTimeUtils.js]: 2 and '0' - magic numbers

**Review by ximet — APPROVED** [review]

**Pull request merged**